test/E2ETests/E2ETests/AspNetCore/CancellationEndToEndTests.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.Functions.Tests;
+using Microsoft.Azure.Functions.Tests.E2ETests;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.Azure.Functions.Worker.E2ETests.AspNetCore
+{
+    public class CancellationEndToEndTests : IClassFixture<CancellationEndToEndTests.TestFixture>
+    {
+        private readonly TestFixture _fixture;
+
+        public CancellationEndToEndTests(TestFixture fixture, ITestOutputHelper testOutputHelper)
+        {
+            _fixture = fixture;
+            _fixture.TestLogs.UseTestLogger(testOutputHelper);
+        }
+
+        [IgnoreOnNetFxTestRunTheory]
+        [InlineData("HttpWithCancellationTokenNotUsed", "Work completed.", "Succeeded")]
+        [InlineData("HttpWithCancellationTokenIgnored", "TaskCanceledException: A task was canceled", "Failed")]
+        [InlineData("HttpWithCancellationTokenHandled", "Request was cancelled.", "Succeeded")]
+        public async Task HttpTriggerFunctions_WithCancellationToken_BehaveAsExpected(string functionName, string expectedMessage, string invocationResult)
+        {
+            using var cts = new CancellationTokenSource();
+
+            try
+            {
+                var task = HttpHelpers.InvokeHttpTrigger(functionName, cancellationToken: cts.Token);
+
+                await Task.Delay(3000);
+                cts.Cancel();
+
+                var response = await task;
+            }
+            catch (Exception)
```

Comment:
Does this test always expect an exception? If so... use `Assert.Throws()` instead (or store the exception and validate it outside the `catch`) so that if it happens to *not* throw, the test will fail. As it is, if an exception is not thrown, the test will pass and we wouldn't notice.

Reply:
Ah that is a great point! Yes, we're always expecting an exception here because we are cancelling the CTS when we invoke the function to mimic RequestAborted -> `await httpClient.SendAsync(request, cancellationToken);` - this will always throw. But it's not really behaviour we need to "test" as this is coming from our helper and not out function/worker code. But agree that we need to assert it throws to not get a false positive